src/Language/DifferentialDatalog/Syntax.hs

```diff
@@ -656,6 +656,72 @@ instance Eq e => Eq (ExprNode e) where
     (==) (ERef _ p1)              (ERef _ p2)                = p1 == p2
     (==) _                        _                          = False
 
+-- Assign rank to constructors; used in the implementation of Ord.
```

Comment:
Why didn't we need to do this previously?

Reply:
We never used expressions as keys in a map and so never had to implement `Ord` for them.